p2p/message.go

```diff
@@ -0,0 +1,76 @@
+package p2p
+
+import (
+	"bytes"
+
+	xdr "github.com/stellar/go-xdr/xdr3"
+)
+
+type Message struct {
+	V  int32
+	V0 MessageV0
+}
+
+func (m *Message) MarshalBinary() ([]byte, error) {
+	b := bytes.Buffer{}
+	_, err := xdr.Marshal(&b, m)
+	if err != nil {
+		return nil, err
+	}
+	return b.Bytes(), nil
+}
+
+func (m *Message) UnmarshalBinary(b []byte) error {
+	_, err := xdr.Unmarshal(bytes.NewReader(b), m)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+type MessageV0 struct {
+	Chain      Chain
+	Body       []byte
```

Comment:
nice, I like that you went with this approach! 💯

Reply:
It was your idea!